File: Clients/Html5Q/tutorials/qframework/GLVertex.js
/*
   Copyright 2012, Telum Slavonski Brod, Croatia.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
   
   This file is part of QFramework project, and can be used only as part of project.
   Should be used for peace, not war :)   
*/

function GLVertex( x,y,z,tu,tv,index) {
	this.x = 0;
	this.y = 0;
	this.z = 0;
	this.tu = 0;
	this.tv = 0;
	this.index = -1;
	this.red = 1;
	this.green = 1;
	this.blue = 1;
	this.alpha = 1;
	


    if (arguments.length == 6)
	{
        this.x = x;
        this.y = y;
        this.z = z;
        this.tu = tu;
        this.tv = tv;
        this.index = index;
    }
}

GLVertex.prototype.equals = function(v) {
	return (this.x == v.x && this.y == v.y && this.z == v.z &&
			v.tu == tu && v.tv == tv && 
			alpha == v.alpha &&
			red == v.red &&
			green == v.green &&
			blue == v.blue);
}

//(FloatBuffer vertexBuffer, ByteBuffer colorBuffer, FloatBuffer textBuffer) {
	GLVertex.prototype.put = function( vertexBuffer,  colorBuffer,  textBuffer) 
	{
        vertexBuffer.push(this.x);
        vertexBuffer.push(this.y);
        vertexBuffer.push(this.z);
		colorBuffer.push(this.red);
		colorBuffer.push(this.green);
		colorBuffer.push(this.blue);
		colorBuffer.push(this.alpha);
        
        textBuffer.push(this.tu);
        textBuffer.push(this.tv);
        
    }

	GLVertex.prototype.put2 = function(vertexBuffer, colorBuffer) 
	{
        vertexBuffer.push(this.x);
        vertexBuffer.push(this.y);
        vertexBuffer.push(this.z);
		colorBuffer.push(this.red);
		colorBuffer.push(this.green);
		colorBuffer.push(this.blue);
		colorBuffer.push(this.alpha);
        
        
    }

    GLVertex.prototype.putText = function(textBuffer,  x, y,  w,  h) 
	{
    	tu1 = this.tu / w + x / w;
    	tv1 = this.tv / h + y / h;
        textBuffer.push(tu1);
        textBuffer.push(tv1);
    }

    GLVertex.prototype.putTextHalf = function(textBuffer,  x, y,  w,  h) 
	{
    	tu1 = this.tu / w + x / w + 1.0;
    	tv1 = this.tv / h + y / h + 1.0;
        textBuffer.push(tu1);
        textBuffer.push(tv1);
    }
	
   

File: Clients/iPhoneQ/Q/QFramework/GLVertex.m
/*
   Copyright 2012, Telum Slavonski Brod, Croatia.

   Licensed under the Apache License, Version 2.0 (the "License");
   you may not use this file except in compliance with the License.
   You may obtain a copy of the License at

       http://www.apache.org/licenses/LICENSE-2.0

   Unless required by applicable law or agreed to in writing, software
   distributed under the License is distributed on an "AS IS" BASIS,
   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
   See the License for the specific language governing permissions and
   limitations under the License.
   
   This file is part of QFramework project, and can be used only as part of project.
   Should be used for peace, not war :)   
*/

#import "GLVertex.h"
#import "FloatBuffer.h"
#import "ByteBuffer.h"
#import "ColorFactory.h"

@implementation GLVertex

@synthesize x;
@synthesize y;
@synthesize z;
@synthesize index;
@synthesize alpha;
@synthesize red;
@synthesize green;
@synthesize blue;
@synthesize tu;
@synthesize tv;

- (id) init
{
	if (self = [super init])
	{
        x = 0;
        y = 0;
        z = 0;
        tu = 0;
        tv = 0;
        index = -1;
        red = 255;
        alpha = 255;
        blue = 255;
        green = 255;        
    }
    return self;
}

- (id) initWithCords:(GLfloat)ax y:(GLfloat)ay z:(GLfloat)az tu:(GLfloat)atu tv:(GLfloat)atv i:(short)aindex
{
	if (self = [super init])
	{
        x = ax;
        y = ay;
        z = az;
        tu = atu;
        tv = atv;
        index = aindex;
        red = 255;
        alpha = 255;
        blue = 255;
        green = 255;                
    }
    return self;
}

- (bool) equals:(GLVertex*) v 
{
    return (x == v.x && y == v.y && z == v.z &&
            v.tu == tu && v.tv == tv && 
				alpha == v.alpha &&
            	red == v.red &&
            	green == v.green &&
            	blue == v.blue);
}

- (void) put:(FloatBuffer*)vertexBuffer cb:(ByteBuffer*)colorBuffer tb:(FloatBuffer*) textBuffer {
    [vertexBuffer put:x];
    [vertexBuffer put:y];
    [vertexBuffer put:z];
    

    [colorBuffer put:red];
    [colorBuffer put:green];
    [colorBuffer put:blue];
    [colorBuffer put:alpha];
    
    [textBuffer put:tu];
    [textBuffer put:tv];
    
}


- (void) put:(FloatBuffer*)vertexBuffer cb:(ByteBuffer*)colorBuffer{
    [vertexBuffer put:x];
    [vertexBuffer put:y];
    [vertexBuffer put:z];
    
    [colorBuffer put:red];
    [colorBuffer put:green];
    [colorBuffer put:blue];
    [colorBuffer put:alpha];
    
    
}

- (void) putText:(FloatBuffer*) textBuffer x:(int)ax y:(int)ay w:(int)aw h:(int)ah {
    
    float tu1 = tu / (float)aw + (float)ax / (float)aw;
    float tv1 = tv / (float)ah + (float)ay / (float)ah;
    [textBuffer put:tu1 ];
    [textBuffer put:tv1 ];
    
}
@end

